tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,91 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
             m.setattr("numpy.random.choice", lambda x: x[0])
             samp = similarity.event_to_sample(photon_num, count, modes_dim)
         assert samp[0] == count
+
+
+orbits = [
+    [(1, 1, 2), 4, 12],
+    [(1, 1), 4, 6],
+    [(1, 2, 3), 4, 24],
+    [(1, 1, 1, 1), 5, 5],
+    [(1, 1, 2), 5, 30],
+    [(1, 2, 3), 5, 60],
+]
+
+
+@pytest.mark.parametrize("orbit, max_photon, expected", orbits)
+def test_orbit_cardinality(orbit, max_photon, expected):
+    """Test if function ``strawberryfields.gbs.similarity.orbit_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+    print('orbit = ', orbit)
+    print('max_photon = ', max_photon)
+    print('expected = ', expected)
+
+    assert similarity.orbit_cardinality(list(orbit), max_photon) == expected
+
+
+events = [
+    [5, 3, 6, 216],
+    [6, 3, 6, 336],
+    [5, 2, 6, 126],
+    [5, 3, 7, 413],
+    [6, 3, 7, 728],
+    [5, 2, 7, 266],
+]
+
+
+@pytest.mark.parametrize("photons, max_count, modes, expected", events)
+def test_event_cardinality(photons, max_count, modes, expected):
+    """Test if function ``strawberryfields.gbs.similarity.event_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    assert similarity.event_cardinality(photons, max_count, modes) == expected
+
+
+class TestPOrbitMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.p_orbit_mc.``"""
+
+    def test_mean_computation_orbit(self, monkeypatch):
+        """Tests if the calculation of the sample mean is performed correctly. The test
+        monkeypatches the fock_prob function so that the probability is the same for each sample and
+        is equal to 1/5, i.e., one over the number of samples in the orbit [1,1,1,1] for 5 modes."""
+        graph = nx.complete_graph(5)
+        with monkeypatch.context() as m:
+            m.setattr(
+                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                lambda *args, **kwargs: 0.2,
+            )
+
+            assert np.allclose(similarity.p_orbit_mc(graph, [1, 1, 1, 1]) - 1.0)
```

Comment:
```suggestion
            assert np.allclose(similarity.prob_orbit_mc(graph, [1, 1, 1, 1]) - 1.0)
```

Reply:
Is this the correct use of `allclose`? https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html
Same for line 286

Reply:
No, should be
```suggestion
            assert np.allclose(similarity.p_orbit_mc(graph, [1, 1, 1, 1]), 1.0)
```